docs/DeveloperGuide.md

```diff
@@ -1,29 +1,693 @@
+This feature's objective is to give users the ability to stay within their financial goals and avoid overspending.
 
-## Product scope
-### Target user profile
+This feature is orchestrated by `ListBudgetCommand` and `SetBudgetCommand`, which are initialised by the `Parser` 
+class. Below is a description of the key class attributes and methods involved in the budget setting and listing 
+process:
+
+##### Class Attributes for `SetBudgetCommand`:
+| Class Attribute | Variable Type | Relevance                                                           |
+|-----------------|---------------|---------------------------------------------------------------------|
+| expenseList     | ExpenseList   | Object containing the list of expenses to check against set budgets |
+| category        | String        | The category for which the budget is being set                      |
+| budget          | double        | The budget amount to be set for the category                        |
+
+##### Class Attributes for `ListBudgetCommand`:
+| Class Attribute | Variable Type | Relevance                                                           |
+|-----------------|---------------|---------------------------------------------------------------------|
+| expenseList     | ExpenseList   | Object containing the list of expenses to check against set budgets |
+
+
+Upon the call of the `execute()` method in `BudgetBuddy` using `command.execute()`, `SetBudgetCommand` will update the 
+budget in `ExpenseList` using `setBudget`. Similarly, `ListBudgetCommand` will fetch and display all categories with 
+their budgets using `getBudgets`, and highlight those that are above the set budget.
+
+##### Key Methods used from `ExpenseList`
+| Method                      | Return Type   | Relevance                                                          |
+|-----------------------------|---------------|--------------------------------------------------------------------|
+| setBudget(category, budget) | void          | Sets or updates the budget for a given category in the ExpenseList |
+| getBudgets()                | List<Budget>  | Retrieves the list of all budgets set                              |
+
+
+#### Sequence diagrams
+
+##### Setting a Budget
+The following UML Sequence diagram shows how `SetBudgetCommand` works when a user sets a budget for a category in the
+following format: `set budget c/<Category> b/<Amount>`
+
+<img alt="SetBSD.png" height="400" src="SetBSD.png" width="700"/>
+
+
+##### Printing Budgets
+The following UML Sequence diagram shows how `ListBudgetCommand` works when a user checks the budget status with the
+command: `budget print`
+
+<img alt="ListBSD2.png" height="400" src="ListBSD2.png" width="700"/>
+
+#### Class diagram
+The class diagram below outlines the relationships between the classes involved in the Budget Management feature:
+
+<img alt="ClassDiagram2.png" height="800" src="ClassDiagram2.png" width="700"/>
```

Comment:
should the green dots in front of the methods be positive or negative signs instead?